fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankIndexMaintainerFactory.java

```diff
@@ -65,24 +65,24 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
 
             @Override
             public void validateChangedOptions(@Nonnull Index oldIndex, @Nonnull Set<String> changedOptions) {
-                // Allow changing from unspecified to the default (or vice versa), but not otherwise.
-                if (changedOptions.contains(IndexOptions.RANK_NLEVELS)) {
-                    int oldLevels = RankedSetIndexHelper.getNLevels(oldIndex);
-                    int newLevels = RankedSetIndexHelper.getNLevels(index);
-                    if (oldLevels != newLevels) {
-                        throw new MetaDataException("rank levels changed",
-                                LogMessageKeys.INDEX_NAME, index.getName());
+                if (!changedOptions.isEmpty()) {
```

Comment:
I disallowed changing duplicates, in keeping with the conservative nature of the existing checks.

Reply:
That sounds right. I'd be tempted to say that even the most liberal of changes shouldn't allow that particular option to change ever (because the data won't be correct if you switch, even if it's understandable).

Reply:
Hmm. There are, I think, two points of view:
* The option defines a fundamental characteristic of the skip list, viz., whether the finest level has counts greater than one.
* The option refines the behavior of the `add` method to either maintain counts or return `false` for things already present.

Which of those makes more sense also probably informs whether `remove` always removes one occurrence or is similarly influenced by the config option.

Reply:
Interesting. I had kind of been coming from the first point of view. It kind of seems like the second point of view makes sense from the point of view of the skip list itself, though it's harder to see how that is a sensible "index" over data. (In that with duplicate tracking off, it is just a data structure for optimizing "rank without counting duplicates" queries; with duplicate tracking on, it is a data structure for optimizing "rank with duplicates counted" queries; but with it changed half-way through, then I don't know if it answers any reasonable queries other than "what is the state of skip list".)

Reply:
Yes, agreed. I don't have a great (not implementation based) idea of why a user of the index would want to change the answers to the rank / select operations it optimizes.